src/Microsoft.OData.Client/Binding/BindingObserver.cs

```diff
@@ -381,8 +381,8 @@ internal void OnDataServiceCollectionChanged(object collection, NotifyCollection
                     break;
 #endif
 
-                default:
-                    throw new InvalidOperationException(Strings.DataBinding_DataServiceCollectionChangedUnknownActionCollection(eventArgs.Action));
+                default: // No Action

```

Comment:
Shouldn't we just remove the conditional compile statement if we want to not throw an exception for move? 
This is a broader change than just move.

Reply:
Discussed offline

Reply:
I believe that, right now, Move is the only enum value that we don't have a conditional for. So the broader change would only be if .NET added new enum values in the future, and the change would be that we would ignore the new values rather than throw an exception. If we had a good way to exclude just Move across frameworks, then I do slightly prefer the default of throwing the exception for unknown values, in case a new enum value represents changes that we should be aware of, but if the (hypothetical) new enum values are correctly defined such that they represent states that are distinct from the existing change values (not returning existing values for existing states would be a breaking change on the .NET side), then we should be able to safely ignore them as well. Given the difficulty in precisely determining which targets do/do not support Move, and the relative unlikelihood of defining a new enum value that breaks existing semantics, I'm okay with the change.

---
In reply to: [355600338](https://github.com/OData/odata.net/pull/1601#discussion_r355600338) [](ancestors = 355600338)